### PR TITLE
Add Shear Object enabling Redshift-dependent Shear

### PR DESCRIPTION
**Pull request by mr-superonion**
@ismael-mendoza and I made this pull request for the redshift dependent shear simulation.
1. [This file](https://github.com/mr-superonion/descwl-shear-sims/blob/shear_obj_zshear/descwl_shear_sims/shear.py) contains the shear objects that is used to generate shear for different tests.
2. Unit test for this implementation can be found [here](https://github.com/mr-superonion/descwl-shear-sims/blob/shear_obj_zshear/shear_meas_tests/test_shear_meas_zshear_metacal.py).

### CODE REVIEW

**Comment by mr-superonion** [comment]
@esheldon , @beckermr, Could you please review when you are free?

**Comment by esheldon** [comment]
One general comment:  I would like to keep shear-measurement specific code out of this repo, for example the new code you added `bin/simulate_image.py`

File: descwl_shear_sims/sim.py [hunk 1]
```diff
@@ -52,12 +52,13 @@ def make_sim(
     rng,
     galaxy_catalog,
     coadd_dim,
-    g1,
-    g2,
+    shear_obj,
```
Comment (esheldon) [inline descwl_shear_sims/sim.py:55]: Can we make this somehow backwards compatible?

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +1,9 @@
-stackvana
+stackvana<=0.2023.28
```
Comment (esheldon) [inline requirements.txt:1]: We should discuss this.  Were you having difficulty with new versions?

**Comment by esheldon** [comment]
The code needs to pass flake8.

File: requirements.txt (hunk 2 above)
Comment (mr-superonion) [reply requirements.txt:1]: hmm, at some point it has an error but I forgot the details. Let me test this part...

File: descwl_shear_sims/sim.py (hunk 1 above)
Comment (mr-superonion) [reply descwl_shear_sims/sim.py]: Done, see [this unit test](https://github.com/mr-superonion/descwl-shear-sims/blob/shear_obj_zshear/shear_meas_tests/test_shear_meas_zshear_metacal_compatible.py) for backward compatibility!

**Comment by mr-superonion** [comment]
Hello @esheldon , I checked that the latest `stackvana=0.2023.47` works. The newer version of mamba (`mamba>1.5.1`) seems to have some problems (see [here](https://github.com/mamba-org/mamba/issues/2994)). Therefore, I set `mamba==1.5.1` in the github workflow.

I think this PR is ready to merge.

File: descwl_shear_sims/sim.py [hunk 3]
```diff
@@ -148,6 +151,11 @@ def make_sim(
 
     if se_dim is None:
         se_dim = get_se_dim(coadd_dim=coadd_dim, dither=dither, rotate=rotate)
+    if shear_obj is None:
+        assert isinstance(g1, float) & isinstance(g2, float), \
```
[the remark this answers is not in the document]
Comment (esheldon) [reply descwl_shear_sims/sim.py:155]: another things a user might do is set g1/g2 from elements of numpy arrays which would also fail the test

[the remark this answers is not in the document]
Comment (mr-superonion) [reply descwl_shear_sims/sim.py]: Thanks. revision made.

**Comment by esheldon** [comment]
Thanks for the PR

**Pull request merged**